### PR TITLE
net-im/telegram-desktop: Add musl support

### PR DESCRIPTION
**Pull request by mid-kid**
@gyakovlev

Are these changes to profile/ acceptable? Any tips?

### CODE REVIEW

File: net-im/telegram-desktop/telegram-desktop-3.5.2-r1.ebuild [hunk 1]
```diff
@@ -95,17 +97,20 @@ src_prepare() {
...
+		-DDESKTOP_APP_DISABLE_DBUS_INTEGRATION=$(usex dbus no yes)
 		-DDESKTOP_APP_DISABLE_X11_INTEGRATION=$(usex X no yes)
 		-DDESKTOP_APP_DISABLE_WAYLAND_INTEGRATION=$(usex wayland no yes)
-		-DDESKTOP_APP_DISABLE_DBUS_INTEGRATION=$(usex dbus no yes)
 		-DDESKTOP_APP_DISABLE_SPELLCHECK=$(usex spell no yes)  # enables hunspell (recommended)
 		-DDESKTOP_APP_USE_ENCHANT=$(usex enchant)  # enables enchant and disables hunspell
+
+		# This option is heavily discouraged by upstream.
+		# See files/tdesktop-*-jemalloc-optional.patch
+		-DDESKTOP_APP_DISABLE_JEMALLOC=$(usex jemalloc no yes)
```
Comment (thesamesam) [inline net-im/telegram-desktop/telegram-desktop-3.5.2-r1.ebuild:113]: Consider:
```suggestion
		-DDESKTOP_APP_DISABLE_JEMALLOC=$(usex !jemalloc)
```

Comment (mid-kid) [reply net-im/telegram-desktop/telegram-desktop-3.5.2-r1.ebuild:113]: Ah, I wasn't aware it accepted this syntax. Thanks!

**Comment by gyakovlev** [comment]
lgtm, but untested, no musl desktop around me.
makes sense to force it on glibc too.

**Comment by mid-kid** [comment]
I tested it in a gentoo prefix with musl, and it worked fine, albeit with a few odd crashes when loading certain stickers. I hope that doesn't happen on *real* musl systems as none of the people using telegram on musl thus far could reproduce it.

**Pull request closed without merging**